akka-http-core/src/main/scala/akka/http/scaladsl/model/MediaType.scala

```diff
@@ -286,6 +286,8 @@ object MediaType {
 }
 
 object MediaTypes extends ObjectRegistry[(String, String), MediaType] {
+
+  /** Function used to find a custom media type. Called before the predefined media types. Strings will be lowercase. */
   type FindCustom = (String, String) ⇒ Option[MediaType]
```

Comment:
The public `ParserSettings` API currently only supports setting this via a `Seq[MediaType]`. That's fine for Play's needs for now, since Play only needs to patch `application/json`.

I have written another patch to the `ParserSettings` API that adds support for using a custom function. I will submit that patch separately after this PR has been merged.